### PR TITLE
Report an error when trying to use integrated auth with SqlClient managed SNI enabled.

### PR DESCRIPTION
**Pull request by corivera**
Currently integrated auth causes a failure in DbConnectionPoolGroup when checking for WindowsIdentity-based pooling, which isn't immediately clear is due to using integrated auth. This change makes the failure more clear by throwing an exception when integrated auth options are being set in the first place while using Managed SNI. Once managed integrated auth on Windows is added, these conditionals can be made platform-specific.

### CODE REVIEW

**Comment by corivera** [comment]
CC @saurabh500 @kkurni

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionString.cs [hunk 1]
```diff
@@ -202,6 +202,12 @@ internal SqlConnectionString(string connectionString) : base(connectionString, G
             }
 
             _integratedSecurity = ConvertValueToIntegratedSecurity();
+#if MANAGED_SNI
+            if(_integratedSecurity)
+            {
+                throw new PlatformNotSupportedException("Integrated security is not supported");
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionString.cs]: Put this in the resource file.

File: src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPoolGroup.cs [hunk 2]
```diff
@@ -160,13 +160,6 @@ internal DbConnectionPool GetConnectionPool(DbConnectionFactory connectionFactor
 
                 if (_poolGroupOptions.PoolByIdentity)
                 {
-                    // There is no concept of Windows identity on non-Windows platform. Hence we 
-                    // cannot support Windows authentication.
-                    if (!ADP.IsWindows)
-                    {
-                        throw new PlatformNotSupportedException(SR.GetString(SR.ADP_FeatureNotSupportedOnNonWindowsPlatform, DbConnectionStringKeywords.IntegratedSecurity));
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPoolGroup.cs]: Remove these from resources file if they are not being used

**Comment by saurabh500** [comment]
@dotnet-bot Test Innerloop OSX Release Build and Test please

**Comment by corivera** [comment]
@dotnet-bot Test Innerloop Ubuntu Release Build and Test please

**Comment by saurabh500** [comment]
LGTM

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by corivera** [comment]
@dotnet-bot Test Innerloop Ubuntu Debug Build and Test please

**Comment by corivera** [comment]
@dotnet-bot Test this please

**Comment by corivera** [comment]
@dotnet-bot Test this please

**Pull request merged**